system/jlib/jfile.cpp

```diff
@@ -7947,38 +7933,63 @@ MODULE_INIT(INIT_PRIORITY_STANDARD)
             PlaneAttributesMapElement &element = planeAttributesMap[plane.queryProp("@name")];
             element.first = plane.queryProp("@prefix");
             auto &values = element.second;
-            unsigned __int64 value;
-            value = plane.getPropInt64(("@" + std::string(planeAttributeTypeStrings[BlockedSequentialIO])).c_str(), unsetPlaneAttrValue);
-            values[BlockedSequentialIO] = (unsetPlaneAttrValue != value) ? value * 1024 : value;
-            value = plane.getPropInt64(("@" + std::string(planeAttributeTypeStrings[BlockedRandomIO])).c_str(), unsetPlaneAttrValue);
-            values[BlockedRandomIO] = (unsetPlaneAttrValue != value) ? value * 1024 : value;
-
-            // plane expert settings
-            std::string propName = "expert/@" + std::string(planeAttributeTypeStrings[FileSyncWriteClose]);
-            if (plane.hasProp(propName.c_str()))
-                values[FileSyncWriteClose] = plane.getPropBool(propName.c_str()) ? 1 : 0;
-            else
+            for (unsigned propNum=0; propNum<PlaneAttributeType::PlaneAttributeCount; ++propNum)
             {
-                // temporary (check legacy fileSyncMaxRetrySecs too), purely for short term backward compatibility (see HPCC-xxxx)
-                unsigned __int64 v = plane.getPropInt64("expert/@fileSyncMaxRetrySecs", unsetPlaneAttrValue);
-                // NB: fileSyncMaxRetrySecs==0 is treated as set/enabled
-                values[FileSyncWriteClose] = v != unsetPlaneAttrValue ? 1 : 0;
+                const PlaneAttributeInfo &attrInfo = planeAttributeInfo[propNum];
+                std::string prop;
+                if (attrInfo.isExpert)
+                    prop += "expert/";
+                prop += "@" + std::string(attrInfo.name);
+                switch (attrInfo.type)
+                {
+                    case PlaneAttrType::integer:
+                    case PlaneAttrType::boolean: // handling is same as integer
+                    {
+                        unsigned __int64 value = plane.getPropInt64(prop.c_str(), unsetPlaneAttrValue);
+                        if (unsetPlaneAttrValue != value)
+                        {
+                            if (attrInfo.scale)
```

Comment:
minor: For simplicity I would have coded using a scale of 1 for the booleans and unconditionally multiplied by the scale.

Reply:
I thought about that, chose not to, with thought in mind that might later have strings too perhaps, and would want to only scale if explicitly made sense to.